source/val/validate_type.cpp

```diff
@@ -140,7 +140,24 @@ spv_result_t ValidateTypeVector(ValidationState_t& _, const Instruction* inst) {
   const auto component_index = 1;
   const auto component_id = inst->GetOperandAs<uint32_t>(component_index);
   const auto component_type = _.FindDef(component_id);
-  if (!component_type || !spvOpcodeIsScalarType(component_type->opcode())) {
+  if (component_type) {
+    bool isPointer = component_type->opcode() == spv::Op::OpTypePointer;
+    bool isScalar = spvOpcodeIsScalarType(component_type->opcode());
+
+    if (_.HasExtension(Extension::kSPV_INTEL_masked_gather_scatter) &&
```

Comment:
I think strictly speaking this should be capability gated by the spec:
```suggestion
    if (_.HasCapability(spv::Capability::MaskedScatterGatherINTEL) &&
```

Reply:
I have made the changes to use capability instead of extension